transcom/mymove
# B 19976

## [19976](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A949508)

[Integration Review](https://github.com/transcom/mymove/pull/12676)
## Summary

This task adds the ability for the prime to add a 'Standalone Crate' item to a move. Its functionality is nearly identical to a standard crating item, with the caveat that when the TOO goes to approve the item, the highest dollar amount that can be approved is 1000$. The TOO approval is part of a follow-on task and is not completed here.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Prime Sim:
1. Access the Office application
2. Login to the Prime Simulator
3. Select a move from the table
4. Find a shipment and select 'Add Service Item'
5. Fill the in service item page, and be sure to select 'Standalone Crate' from the 'Service Item Type' dropdown
6. Fill in the rest of the form
7. Submit the form
8. Verify that the Standalone Crate service item was added to the shipment (this can be checked by looking through the service item list and verifying that a DCRTSA Service Item exists in the list. The prime UI does not show details for crating items. Screenshot attached)

Swagger/ Postman:
Swagger URL: http://primelocal:3000/swagger-ui/prime.html#/mtoServiceItem/createMTOServiceItem
Payload: {"item":{"length":4,"width":4,"height":4},"crate":{"length":5,"width":5,"height":5},"moveTaskOrderID":"958776a4-782f-4657-bd29-0e7f5cf5bedc","mtoShipmentID":"67d0fd70-eba3-4455-83b8-ab658bc8099f","modelType":"MTOServiceItemDomesticStandaloneCrating","reason":"i want to","description":"moped","reServiceCode":"DCRTSA"}

1. Navigate to 'createMTOServiceItem' in swagger/postman
2. Copy the payload in (with a moveTaskOrderID that exists on your system and is available to the prime)
3. Execute the action
4. Verify you received a success return code
5. Verify the Service Item was added to the shipment in the DB or Prime UI

### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-05-13 at 8 43 33 AM](https://github.com/transcom/mymove/assets/162157577/6c79b627-1f83-43a7-8721-66c5d6164066)
![Screenshot 2024-05-13 at 8 42 55 AM](https://github.com/transcom/mymove/assets/162157577/b93de46e-f631-4eb8-a785-56788691874e)
![Screenshot 2024-05-15 at 10 05 12 AM](https://github.com/transcom/mymove/assets/162157577/d980ba76-0c5c-4863-9595-65e94a883384)
